### DISCUSSION TITLE
How to use background image in Twig with Tailwindcss?

### DISCUSSION THREAD

**Discussion by karaka200** [tailwindlabs/tailwindcss#9242] [Help] [answered]
I use `Symfony` framework. I want to add an background image to my element with `Tailwindcss` framework.

I use this in `index.html.twig` :

`<div class="w-full bg-[url('{{ asset('images/background.png') }}')]"></div>`
When run `npm run watch` and go to browser, it create `<div class="w-full bg-[url('/images/background.png')]"></div>` but `background.png` not show and `background-image: url(/images/background.png);` was not created in css file.

How can I fix this problem?

**Comment by wongjn** [comment, marked as answer]
This is because Tailwind sees the string source as-is, and sees these class string candidates:
 - `w-full`  
 - `bg-[url('{{`
 - `asset('images/background.png')`
 - `}}')]`

For this kind of value not known at build time, you could use CSS variables with an arbitrary value:
```html
<div class="w-full bg-[image:var(--image)]" style="--image: url('{{ asset('images/background.png') }}')"></div>
```

But then if you are not going to use any variants with it, it might be best to set the background image as a style attribute directly:
```html
<div class="w-full" style="background-image: url('{{ asset('images/background.png') }}')"></div>
```

**Reply by karaka200** [reply]
Thank you.
I use `<div class="w-full bg-[var(--image)]" style="--image: url('{{ asset('images/background.png') }}')"></div>` but not worked. So I have to use `<div class="w-full" style="background-image: url('{{ asset('images/background.png') }}')"></div>`

**Reply by wongjn** [reply]
Ah, yes forgot to add the value type prefix for the first method, so it should be `bg-[image:var(--image)]`.
